### PR TITLE
feat(ui): plugin update detection UI (Phase 2b)

### PR DESCRIPTION
**Pull request by dwalleck**
## Summary

Implements Phase 2b of the plugin update detection rollout — the FE consumer for the wire format Phase 2a (PR #96) shipped.

- **Update indicator + button on PluginCard (BrowseTab)** — `Update → vN` button replaces the `Installed` pill when an update is available; `Update check failed` red pill when scan failed; orange "Update (content changed)" label for ContentChanged drift signal.
- **Status column + Update action on InstalledTab** — new column between Version and Contents shows per-plugin update/failure/up-to-date state. Action cell gets `[Update] [Remove]` with single in-flight mutex (any in-flight action disables both).
- **`RemovePluginResult` reshape consumed** — InstalledTab's Remove toast now uses an inline `<details>` block to itemize per-content-type successes and failures, replacing the previous opaque count message.
- **Failures group by `(remediationClass, marketplace)`** — N stale-cache failures from one marketplace produce one banner ("3 plugins from acme couldn't be checked. Run `kiro-market update`...") instead of N. Per-row pills share a generic "Update check failed" treatment with `kindLabel`-driven `title=` tooltip.
- **`pluginUpdates` Svelte store** at `src/lib/stores/plugin-updates.svelte.ts` — eager scan on project mount + after marketplace fetch (via new `MarketplacesTab.onUpdated` callback). Includes race-condition guards for fast project switching and explicit error/result clearing on transitions.
- **`BannerStack` shared component** extracted from BrowseTab; consumed by both tabs. Uses Svelte 5's `<script generics="K extends string">` to propagate the per-tab `ErrorSource` literal-union type without casts.
- **Vitest setup (narrow scope)** — pure-logic helpers tested in `node` env; no jsdom, no `@testing-library/svelte`. 23 tests covering `remediationClass`, `kindLabel`, `groupFailures`, `formatInstallPluginResult`, `formatRemovePluginResult`. Component-level testing intentionally future scope.

## Design + plan
- Design: `docs/plans/2026-05-05-phase-2b-update-detection-ui-design.md`
- Plan: `docs/plans/2026-05-05-phase-2b-update-detection-ui-plan.md`
- Plan-review record: in the plan's "Plan-review record" section near the end (consolidates the 6-gate review findings + 9 amendments per the project's plan-review-checklist process).

## Architecture decisions worth flagging on review

- **DELIM-aware `update-check<DELIM>` ErrorSource keys** — composite key family for grouped failure banners, follows the existing `pluginKey()` separator convention from `keys.ts`. Initial plan-review pass surfaced a false positive about missing separators (the plan's literal bytes contained `` which Read renders invisibly); implementation uses explicit `${typeof DELIM}` / `${DELIM}` / `DELIM` everywhere.
- **`Extract<PluginAction, ...>` per-tab narrowing** — shared `PluginAction` union (`"install" | "update" | "remove"`) exported from `plugin-updates.ts`; BrowseTab narrows to `"install" | "update"`, InstalledTab narrows to `"remove" | "update"`. Single source of truth for the action discriminator.
- **Project-change state cleanup** (commit `5893c25`) — both tabs explicitly clear banners, pendingPluginActions, and project-scoped fetchErrors keys on `projectPath` transitions to prevent stale state leaking across project switches.
- **Cascade ordering uniform across all action handlers** — `pluginUpdates.refresh(projectPath)` first, then local refresh. Applies to `installWholePlugin`, BrowseTab's `updatePlugin`, InstalledTab's `removePlugin`, and InstalledTab's `updatePlugin`.

## Test plan
- [x] `cargo fmt --all --check`
- [x] `cargo test --workspace`
- [x] `cargo clippy --workspace --tests -- -D warnings`
- [x] `cd crates/kiro-control-center && npm run check` (353 files, 0 errors, 0 warnings)
- [x] `cd crates/kiro-control-center && npm run test:unit` (23 tests)
- [x] `cargo xtask plan-lint` (all 6 gates pass)
- [ ] Manual UI smoke (golden + edge cases: no plugins / scan-in-progress / ContentChanged label / legacy install / marketplace offline / Remove with mixed sub-results)
- [ ] Playwright e2e with `FIXTURE_MARKETPLACE_PATH` set (Task 18 added Status-column-headers test; the toast scenario remains fixture-dependent)

🤖 Generated with [Claude Code](https://claude.com/claude-code)

### CODE REVIEW

File: crates/kiro-control-center/src/lib/components/BrowseTab.svelte [hunk 1]
```diff
@@ -847,7 +924,52 @@
...
+      } else {
+        installError = `Update failed for ${plugin}: ${result.error.message}`;
+      }
+    } catch (e) {
+      const reason = e instanceof Error ? e.message : String(e);
+      installError = `Update failed for ${plugin}: ${reason}`;
+    } finally {
+      pendingPluginActions.delete(key);
     }
   }
```
[the remark this answers is not in the document]
Comment (dwalleck) [reply crates/kiro-control-center/src/lib/components/BrowseTab.svelte]: Addressed in 5728869: `installWholePlugin` was collapsed with `updatePlugin` into a single `runPluginInstall(marketplace, plugin, mode)` that delegates result formatting to `formatInstallPluginResult`. The `mode` parameter is typed as `Extract<PluginAction, "install" | "update">` so the same call dispatches both flows with mode-dependent banner copy and `force` flag. Net ~40 lines saved, drift risk removed.

File: crates/kiro-control-center/src/lib/stores/plugin-updates.ts [hunk 2]
```diff
@@ -0,0 +1,125 @@
...
+ *  ordering of group keys; plugin order within a group is input order.
+ */
+export function groupFailures(failures: PluginUpdateFailure[]): FailureGroup[] {
+  const map = new Map<string, FailureGroup>();
+  for (const f of failures) {
+    const cls = remediationClass(f.kind);
+    // Composite key — a literal "<remediation>:<marketplace>" suffices
+    // because remediationClass values never contain ":" and marketplace
+    // names go through a backend newtype that forbids control chars.
+    const groupKey = `${cls}:${f.marketplace}`;
```
[the remark this answers is not in the document]
Comment (dwalleck) [reply crates/kiro-control-center/src/lib/stores/plugin-updates.ts]: Addressed in 52f2406. Switched the `groupFailures` composite key to `${cls}${DELIM}${marketplace}` and imported `DELIM` from `$lib/keys`. Worth flagging that the prior comment claimed marketplace names "forbid `:` via newtype," but `validate_name` in `kiro-market-core/src/validation.rs:550` only rejects control chars, path separators, `..`, leading/trailing whitespace, and Windows-reserved names — `:` was actually allowed. There was no runtime collision risk (RemediationClass is a fixed-set string union and Map equality is exact-string), but the `:` was inconsistent with the ErrorSource convention and the comment was factually wrong. Added a regression test asserting `groupFailures` doesn't collide on marketplace names containing `:`.

**Comment by dwalleck** [comment]
@copilot resolve the merge conflicts in this pull request
